api/v1alpha1/serviceintentions_types.go

```diff
@@ -296,12 +333,34 @@ func (in IntentionPermissions) validate(path *field.Path) field.ErrorList {
 
 func (in *IntentionHTTPPermission) validate(path *field.Path) field.ErrorList {
 	var errs field.ErrorList
+	if (in.PathExact != "" && in.PathPrefix != "") || (in.PathExact != "" && in.PathRegex != "") || (in.PathPrefix != "" && in.PathRegex != "") {
+		errs = append(errs, field.Invalid(path, in, `At most only one of pathExact, pathPrefix, or pathRegex may be configured.`))
```

Comment:
The other places we json encode `in`.

Reply:
I did from the docs. But when I looked at the Consul code, they were using headers from the HTTP library so I switched to using those and added a validation to ensure one doesnt repeat a methods as well.

Reply:
Sorry, I mean instead of passing `in` to the field error that ends up getting encoded in error messages as the golang struct in other locations we're json encoding `in` first

Reply:
whoops..this was the response to the question about the accepted header fields. sorry.

yes yes, i updated it to json encode `in` and pass that into the error field 😄